python/mlcroissant/mlcroissant/_src/operation_graph/execute.py

```diff
@@ -124,3 +132,37 @@ def read_all_files():
                 "An error occured during the streaming generation of the dataset, more"
                 f" specifically during the operation {operation}"
             ) from e
+
+
+def execute_operations_in_beam(
+    pipeline: beam.Pipeline, record_set: str, operations: Operations
+):
+    """Executes operations in Beam.
+
+    This is only possible for streamable datasets, i.e. datasets that result from
+    executing a linear sequence of operations. If there are branches we'd need a more
```

Comment:
Does this include e.g. joins too?

Reply:
This would be important to know because if yes, IIUC then we would have a problem with datasets with splits?

Reply:
Done.